File: urbanairship-core/src/main/java/com/urbanairship/app/ApplicationListener.java
/* Copyright Airship and Contributors */

package com.urbanairship.app;

/**
 * Listener for application foreground and backgrounds.
 */
public interface ApplicationListener {

    /**
     * Called when the app is foregrounded.
     *
     * @param milliseconds Time in milliseconds when the foreground occurred.
     */
    void onForeground(long milliseconds);

    /**
     * Called when the app is backgrounded.
     *
     * @param milliseconds Time in milliseconds when the background occurred.
     */
    void onBackground(long milliseconds);

}

File: urbanairship-core/src/main/java/com/urbanairship/app/ApplicationListener.kt
/* Copyright Airship and Contributors */
package com.urbanairship.app

import java.time.Instant

/**
 * Listener for application foreground and backgrounds.
 */
public interface ApplicationListener {

    /**
     * Called when the app is foregrounded.
     *
     * @param timestamp The time when the foreground occurred.
     */
    public fun onForeground(timestamp: Instant)

    /**
     * Called when the app is backgrounded.
     *
     * @param timestamp The time when the background occurred.
     */
    public fun onBackground(timestamp: Instant)
}
